src/mrpro/algorithms/optimizers/_fista.py

```diff
@@ -113,46 +54,70 @@ def fista_with_backtracking(
         convex, non-smooth functional with computable proximal map
     initial_value
         initial value for the solution of the algorithm
-    stepsize_initial
-        initial stepsize for gradient step
+    stepsize
+        stepsize needed in the gradient step, is constant throughout all
+        iterations
+    reg_parameter
+        regularization parameter that multiplies g
     max_iterations, optional
-        maximal number of iterations
-    
+        number of iterations
+    backtrack_factor
+        must be <=1. if <1., Backtracking rule for stepsize following https://www.ceremade.dauphine.fr/~carlier/FISTA
+        is used
+
     Returns
     -------
         an approximate solution of the minimization problem
     """
-    x = initial_value
-    y = x
-    t = 1.
-    stepsize_old = stepsize_initial
-    
-    for iteration in range(max_iterations):
-        # updte the stepsize 
-        stepsize_prox = backtracking_stepsize_rule(f,g,stepsize_old, y, 
-                                                   reg_parameter)
-        
-        # calculate the proximal step
-        x_update = g.prox(y - stepsize_prox * torch.autograd.grad(f(y), y), 
-                          reg_parameter)
-        
-        # update the stepsize
-        t_update = (1 + math.sqrt(1 + 4*t**2))/2
-        
+    backtracking = not math.isclose(backtrack_factor, 1)
+    complex_input = torch.is_complex(initial_value)
+    x_old = initial_value
+    y = initial_value
+    t_old = 1.0
+
+    for _ in range(max_iterations):
+        while stepsize > 1e-30:
+            f_y = f(y)
+            # calculate the proximal gradient step
+            gradient = grad(f_y, y)
+            x = g.prox(y - stepsize * gradient, reg_parameter * stepsize)
+
+            if not backtracking:
+                # no need to check stepsize, continue to next iteration
+                break
+            difference = x - y
+            Q = f_y + 1 / (2 * stepsize) * difference.abs().square().sum()
+            if complex_input:
```

Comment:
pretty sure this is not requiered, you can always do .real, even on real numbers.

Reply:
ah nice!